spot_driver/spot_driver/spot_ros2.py

```diff
@@ -1442,6 +1556,272 @@ def handle_set_volume(self, request: SetVolume.Request, response: SetVolume.Resp
+            return response
+        except Exception as e:
+            response.success = False
+            response.message = f"Error: {e}"
+            return response
+
+    def handle_delete_logpoint(
+        self, request: DeleteLogpoint.Request, response: DeleteLogpoint.Response
+    ) -> DeleteLogpoint.Response:
+        """Ros service handler for deleting a logpoint from SpotCAM"""
+        if self.spot_cam_wrapper is None:
+            response.success = False
+            response.message = "Spot CAM has not been initialized"
+            return response
+        try:
+            self.spot_cam_wrapper.media_log.delete_logpoint(request.name)
+            response.success = True
+            response.message = "Success"
+            return response
+        except Exception as e:
+            response.success = False
+            response.message = f"Error: {e}"
+            return response
+
+    def handle_store_logpoint(
+        self, request: StoreLogpoint.Request, response: StoreLogpoint.Response
+    ) -> StoreLogpoint.Response:
+        """Ros service handler for storing current camera data as a logpoint on SpotCAM"""
+        if self.spot_cam_wrapper is None:
+            response.success = False
+            response.message = "Spot CAM has not been initialized"
+            return response
+
+        try:
+            tag = None if request.tag == "" else request.tag
+            camera_name = SpotCamCamera(request.name)  # Silly but don't want to modify cam wrapper.
+            proto_logpoint = self.spot_cam_wrapper.media_log.store(camera_name, tag)
+            conv.convert_proto_to_bosdyn_msgs_logpoint(proto_logpoint, response.logpoint)
+            response.success = True
+            response.message = "Success"
+            return response
+        except Exception as e:
+            response.success = False
+            response.message = f"Error: {e}"
+            return response
+
+    def handle_tag_logpoint(self, request: TagLogpoint.Request, response: TagLogpoint.Response) -> TagLogpoint.Response:
+        """Ros service handler for adding a tag to a logpoint on SpotCAM"""
+        if self.spot_cam_wrapper is None:
+            response.success = False
+            response.message = "Spot CAM has not been initialized"
+            return response
+        try:
+            self.spot_cam_wrapper.media_log.tag(request.name, request.tag)
+            response.success = True
+            response.message = "Success"
+            return response
+        except Exception as e:
+            response.success = False
+            response.message = f"Error: {e}"
+            return response
+
+    def handle_get_led_brightness(
+        self, request: GetLEDBrightness.Request, response: GetLEDBrightness.Response
+    ) -> GetLEDBrightness.Response:
+        """Ros service handler for getting the current brightness of the Spot CAM onboard LEDs"""
+        if self.spot_cam_wrapper is None:
```

Comment:
I hate to nitpick here, but is there a way we can do this without copy pasting this block?